automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/cluster/ClusterAutopauseSpec.scala

```diff
@@ -0,0 +1,28 @@
+package org.broadinstitute.dsde.workbench.leonardo.cluster
+
+import org.broadinstitute.dsde.workbench.leonardo.{ClusterFixtureSpec, ClusterStatus, Leonardo, LeonardoTestUtils}
+import org.scalatest.time.{Minutes, Seconds, Span}
+import org.scalatest.DoNotDiscover
+
+@DoNotDiscover
+class ClusterAutopauseSpec extends ClusterFixtureSpec with LeonardoTestUtils {
+
+  "patch autopause then pause properly" in { clusterFixture =>
+    Leonardo.cluster
+      .get(clusterFixture.cluster.googleProject, clusterFixture.cluster.clusterName)
+      .autopauseThreshold shouldBe 0
```

Comment:
what's the behavior when autopause is 0? 🤔 
Can we create a cluster with short autopause threshold directly instead of patching it afterwards?

Reply:
The logic is 

```scala
private def calculateAutopauseThreshold(autopause: Option[Boolean], autopauseThreshold: Option[Int]): Int =
    autopause match {
      case None =>
        autoFreezeConfig.autoFreezeAfter.toMinutes.toInt
      case Some(false) =>
        autoPauseOffValue
      case _ =>
        if (autopauseThreshold.isEmpty) autoFreezeConfig.autoFreezeAfter.toMinutes.toInt
        else Math.max(autoPauseOffValue, autopauseThreshold.get)
    }
```

It's sort of complicated and I couldn't find unit or integration tests that check all of the possibilities 🤷‍♂

Reply:
it does nothing when its 0 https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/resources/swagger/api-docs.yaml#L980

We can but the patch takes negligible time. I figure since each test is so expensive, its good to test that patching it works as well in the same test.

Reply:
ah, I found a test in unit tests for the patch. I'll update this, and add a test for the calculateAutopauseThreshold logic